fhir-questionnaire-generation-pipeline/data_ingestion_pipeline/pdf_to_md_service/ftp_utils.py

```
@@ -0,0 +1,113 @@
+        # Change to the pdf directory
+        ftp.cwd('/pdf')
+        # Prepare the PDF filename
+        pdf_filename = f"{file_name}.pdf"
+        # Create a temporary file to store the downloaded PDF
+        with tempfile.NamedTemporaryFile(delete=False, suffix='.pdf') as temp_file:
+            # Download the PDF file from FTP server
+            ftp.retrbinary(f'RETR {pdf_filename}', temp_file.write)
+            temp_file_path = temp_file.name
+        logger.info(f"Successfully downloaded PDF file for {file_name} from FTP server")
+        return temp_file_path
+    except Exception as e:
```

Ruff (0.15.6)

[warning] Do not catch blind exception: `Exception`

(BLE001)